### ISSUE TITLE
Backward compatibility with `geopandas<1.0.0`

### ISSUE THREAD

**Issue by qpoterek** [Deltares/pandamesh#44]
The latest versions of `pandamesh` use `geopandas>=1.0.0`, namely for leveraging the `polygonize` available on GeoDataFrames. However, several changes, including the preprocessor, would be of great interest to our workflows, which otherwise still rely on libraries that depend on `geopandas<1.0.0`.

Running `pandamesh==0.2.0` with `geopandas==0.14.4` and `python==3.9.18` results in the following error when trying to mesh a GeoDataFrame with the TriangleMesher:

```shell
An error occurred: 'GeoDataFrame' object has no attribute 'polygonize'.
```

By any chance, would it be possible to circumvent this breaking change and introduce backward compatibility with `geopandas<1.0.0`?

Thank you for your assistance!

**Comment by Huite** [comment]
Hi @qpoterek

I've just pushed https://github.com/Deltares/pandamesh/commit/3ffc6c7452ea8f2353114b90c88ceb62a693ac9b
I've set geopandas to 0.14 locally, ran the tests and replace some geopandas/geoseries methods by shapely functions.

If you have some time, might be worthwhile to checkout `main` branch and see if everything runs properly for you -- I can do a new release promptly then.

**Comment by qpoterek** [comment]
Hello @Huite! Sorry for the late answer, and thank you very much for your feedback and swift work on this issue. I was able to test these changes on different versions of Python, down to 3.9, and I am pleased to confirm that the new workflow works very well!

Also, very nice to see someone else use `pixi` ;)

Again, thank you, and I think you can go forward with the new release. I will let the issue opened for the moment and let you close it at your own discretion if that's okay.

**Comment by Huite** [comment]
I've just released 0.2.2 on pypi: https://pypi.org/project/pandamesh/0.2.2/

Conda-forge release will be up soon.

(And yes, pixi is wonderful!)
